test/test_roundtrip.py

```diff
@@ -25,6 +28,8 @@
 
 
 SKIP = [
+    ('xml', 'test/n3/n3-writer-test-29.n3'),  # has predicates that cannot be shortened to strict qnames
+    ('application/rdf+xml', 'test/n3/n3-writer-test-29.n3'),  # has predicates that cannot be shortened to strict qnames
```

Comment:
huh?

Reply:
For example there is no valid way to turn `http://example.org/1#1` into a qname.
`ValueError: Can't split 'http://example.org/1#1'` is the error producing when running the test.
https://github.com/RDFLib/rdflib/blob/7abed6165d4ad235d41d1ddfb9dbbc2c64d6df0b/test/n3/n3-writer-test-29.n3#L17

Working on this I did find a new bug though!

Reply:
Test output when not skipped
```
======================================================================
ERROR: test.test_roundtrip.test_cases(('nt', 'xml', 'test/nt/qname-02.nt'),)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tom/git/rdflib/rdflib/namespace.py", line 400, in compute_qname_strict
    namespace, name = split_uri(uri, NAME_START_CATEGORIES)
  File "/home/tom/git/rdflib/rdflib/namespace.py", line 600, in split_uri
    raise ValueError("Can't split '{}'".format(uri))
ValueError: Can't split 'http://example.org/#'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/tom/git/rdflib/test/test_roundtrip.py", line 45, in roundtrip
    s = g1.serialize(format=testfmt)
  File "/home/tom/git/rdflib/rdflib/graph.py", line 932, in serialize
    serializer.serialize(stream, base=base, encoding=encoding, **args)
  File "/home/tom/git/rdflib/rdflib/plugins/serializers/rdfxml.py", line 68, in serialize
    bindings = list(self.__bindings())
  File "/home/tom/git/rdflib/rdflib/plugins/serializers/rdfxml.py", line 35, in __bindings
    prefix, namespace, name = nm.compute_qname_strict(predicate)
  File "/home/tom/git/rdflib/rdflib/namespace.py", line 404, in compute_qname_strict
    raise ValueError(message)
ValueError: This graph cannot be serialized to a strict format because there is no valid way to shorten http://example.org/#

======================================================================
ERROR: test.test_roundtrip.test_n3(('n3', 'xml', 'test/n3/n3-writer-test-29.n3'),)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tom/git/rdflib/rdflib/namespace.py", line 400, in compute_qname_strict
    namespace, name = split_uri(uri, NAME_START_CATEGORIES)
  File "/home/tom/git/rdflib/rdflib/namespace.py", line 600, in split_uri
    raise ValueError("Can't split '{}'".format(uri))
ValueError: Can't split 'http://example.org/1'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/tom/git/rdflib/test/test_roundtrip.py", line 45, in roundtrip
    s = g1.serialize(format=testfmt)
  File "/home/tom/git/rdflib/rdflib/graph.py", line 932, in serialize
    serializer.serialize(stream, base=base, encoding=encoding, **args)
  File "/home/tom/git/rdflib/rdflib/plugins/serializers/rdfxml.py", line 68, in serialize
    bindings = list(self.__bindings())
  File "/home/tom/git/rdflib/rdflib/plugins/serializers/rdfxml.py", line 35, in __bindings
    prefix, namespace, name = nm.compute_qname_strict(predicate)
  File "/home/tom/git/rdflib/rdflib/namespace.py", line 404, in compute_qname_strict
    raise ValueError(message)
ValueError: This graph cannot be serialized to a strict format because there is no valid way to shorten http://example.org/1


```